server/api/crud/pagination_cache.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import datetime
+import typing
+
+import sqlalchemy.orm
+
+import mlrun.common.schemas
+import mlrun.config
+import mlrun.utils.singleton
+import server.api.crud
+import server.api.utils.singletons.db
+
+
+class PaginationCache(metaclass=mlrun.utils.singleton.Singleton):
+    @staticmethod
+    def store_pagination_cache_record(
+        session: sqlalchemy.orm.Session,
+        user: str,
+        method: typing.Callable,
+        current_page: int,
+        kwargs: dict,
+    ):
+        db = server.api.utils.singletons.db.get_db()
+        return db.store_paginated_query_cache_record(
+            session, user, method.__name__, current_page, kwargs
+        )
+
+    @staticmethod
+    def get_pagination_cache_record(session: sqlalchemy.orm.Session, key: str):
+        db = server.api.utils.singletons.db.get_db()
+        return db.get_paginated_query_cache_record(session, key)
+
+    @staticmethod
+    def list_pagination_cache_records(
+        session: sqlalchemy.orm.Session,
+        key: str = None,
+        user: str = None,
+        function: str = None,
+        last_accessed_before: datetime = None,
+        order_by: typing.Optional[
+            mlrun.common.schemas.OrderType
+        ] = mlrun.common.schemas.OrderType.desc,
+    ):
+        db = server.api.utils.singletons.db.get_db()
+        return db.list_paginated_query_cache_record(
+            session, key, user, function, last_accessed_before, order_by
+        )
+
+    @staticmethod
+    def delete_pagination_cache_record(session: sqlalchemy.orm.Session, key: str):
+        db = server.api.utils.singletons.db.get_db()
+        db.delete_paginated_query_cache_record(session, key)
+
+    @staticmethod
+    def cleanup_pagination_cache(session: sqlalchemy.orm.Session):
+        db = server.api.utils.singletons.db.get_db()
+        db.list_paginated_query_cache_record(session, as_query=True).delete()
+        session.commit()
```

Comment:
why do you need to cmmit here? why not callee commit on the session?

Reply:
I don't understand the question. I am calling commit on the session

Reply:
why do you commit here? why not let the callee of `cleanup_pagination_cache` or the session manager commit once done with the session?